src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -84,6 +91,15 @@ public synchronized static OpenShiftClient getOpenShiftClient() {
     return openShiftClient;
   }
 
+  // Get the current OpenShiftClient and configure to use the current Oauth token.
+  public synchronized static OpenShiftClient getAuthenticatedOpenShiftClient() {
+    if(openShiftClient != null){
+      openShiftClient.getConfiguration().setOauthToken(CredentialsUtils.getCurrentToken());
```

Comment:
Have you tested without a explicitly configured credential, where the k8s-client can use the serviceaccount token mounted in the pod?  It seems to me that this will override the token set there during the construction of the client's Configuration via io.fabric8.kubernetes.client.Config.tryServiceAccount(Config) when the client is first constructed.

I think you should only call `setOauthToken` if `CredentialsUtils.getCurrentToken` returns a string whose length is greater than zero (i.e. it found a credential).